apps/canary-atomy-api/src/Service/Settings/Adapters/SettingsProvider.php

```
@@ -0,0 +1,92 @@
+        $this->validateTenant($tenantId);
+        return $this->settingRepository->has($key);
+    }
+
+    public function getSettingsByCategory(string $category, string $tenantId): array
+    {
+        $this->validateTenant($tenantId);
+        // Using prefix as a proxy for category
+        return $this->settingRepository->getByPrefix($category);
+    }
+
+    public function update(string $key, mixed $value, ?string $tenantId = null, ?string $userId = null): void
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$userId'. (undefined)

(UnusedFormalParameter)